haystack/utils/deserialization.py

```diff
@@ -68,7 +68,4 @@ def deserialize_chatgenerator_inplace(data: Dict[str, Any], key: str = "chat_gen
     except ImportError as e:
         raise DeserializationError(f"Class '{serialized_chat_generator['type']}' not correctly imported") from e
 
-    if not hasattr(chat_generator_class, "from_dict"):
-        raise DeserializationError(f"Class '{chat_generator_class}' does not have a 'from_dict' method")
-
-    data[key] = chat_generator_class.from_dict(serialized_chat_generator)
+    data[key] = component_from_dict(chat_generator_class, serialized_chat_generator, "chat_generator")
```

Comment:
I think this should reuse the `key` variable right?
```suggestion
    data[key] = component_from_dict(chat_generator_class, serialized_chat_generator, key)
```

Reply:
Looking at [`component_from_dict` implementation](https://github.com/deepset-ai/haystack/blob/45aa9608b54497ba7d8f566bd4dceeb6ae658d4e/haystack/core/serialization.py#L134), it seems that the `name` (3rd keyword argument - mandatory) is only used in relation to deserialization callbacks (which we don't use in this case), so I believe that using a fixed value is Ok.

To be more explicit, I've changed this PR to use keyword arguments.

Reply:
okay sounds good!